.hermes/plans/2026-04-10_220000-files_scanned-truncation-fix-plan.md

```
@@ -0,0 +1,82 @@
+
+## Proposed Fix
+
+### Step 1 — Add clamping at the overflow point
+
+In `crates/diffguard-domain/src/evaluate.rs` line ~312, change:
+```rust
+files_scanned: files_seen.len() as u32,
+```
+to:
+```rust
+files_scanned: files_seen.len().min(u32::MAX as usize) as u32,
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)